hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/slowlog/TestSlowLogRecorder.java

```diff
@@ -0,0 +1,312 @@
+    confirmPayloadParams(1, 142, slowLogPayloads);
+    confirmPayloadParams(2, 143, slowLogPayloads);
+    confirmPayloadParams(3, 144, slowLogPayloads);
+    confirmPayloadParams(4, 145, slowLogPayloads);
+    confirmPayloadParams(5, 146, slowLogPayloads);
+    confirmPayloadParams(6, 147, slowLogPayloads);
+    confirmPayloadParams(7, 148, slowLogPayloads);
+    confirmPayloadParams(8, 149, slowLogPayloads);
+    confirmPayloadParams(9, 150, slowLogPayloads);
+    confirmPayloadParams(10, 151, slowLogPayloads);
+    confirmPayloadParams(11, 152, slowLogPayloads);
+    confirmPayloadParams(12, 153, slowLogPayloads);
+    confirmPayloadParams(13, 154, slowLogPayloads);
+
+    boolean isRingBufferCleaned = slowLogRecorder.clearSlowLogPayloads();
+    Assert.assertTrue(isRingBufferCleaned);
+    LOG.debug("cleared the ringbuffer of Online Slow Log records");
+    slowLogPayloads = slowLogRecorder.getSlowLogPayloads(request);
+    // confirm ringbuffer is empty
+    Assert.assertEquals(slowLogPayloads.size(), 0);
+  }
+
+  @Test
+  public void testOnlineSlowLogWithDefaultDisableConfig() throws Exception {
+    Configuration conf = HBASE_TESTING_UTILITY.getConfiguration();
+    conf.unset(HConstants.SLOW_LOG_BUFFER_ENABLED_KEY);
+    slowLogRecorder = new SlowLogRecorder(conf);
+    AdminProtos.SlowLogResponseRequest request =
+      AdminProtos.SlowLogResponseRequest.newBuilder().build();
+    Assert.assertEquals(slowLogRecorder.getSlowLogPayloads(request).size(), 0);
+    LOG.debug("Initially ringbuffer of Slow Log records is empty");
+
+    for (int i = 0; i < 300; i++) {
+      SlowLogPayload slowLogPayload = getSlowLogPayload(i + 1);
+      slowLogRecorder.addSlowLogPayload(slowLogPayload);
+    }
+
+    Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
+
+    List<SlowLogPayload> slowLogPayloads = slowLogRecorder.getSlowLogPayloads(request);
+    Assert.assertEquals(slowLogPayloads.size(), 0);
+  }
+
+  @Test
+  public void testOnlineSlowLogWithDisableConfig() throws Exception {
+    Configuration conf = HBASE_TESTING_UTILITY.getConfiguration();
+    conf.setBoolean(HConstants.SLOW_LOG_BUFFER_ENABLED_KEY, false);
+    slowLogRecorder = new SlowLogRecorder(conf);
+    AdminProtos.SlowLogResponseRequest request =
+      AdminProtos.SlowLogResponseRequest.newBuilder().build();
+    Assert.assertEquals(slowLogRecorder.getSlowLogPayloads(request).size(), 0);
+    LOG.debug("Initially ringbuffer of Slow Log records is empty");
+
+    for (int i = 0; i < 300; i++) {
+      SlowLogPayload slowLogPayload = getSlowLogPayload(i + 1);
+      slowLogRecorder.addSlowLogPayload(slowLogPayload);
+    }
+
+    Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
+
+    List<SlowLogPayload> slowLogPayloads = slowLogRecorder.getSlowLogPayloads(request);
+    Assert.assertEquals(slowLogPayloads.size(), 0);
+    conf.setBoolean(HConstants.SLOW_LOG_BUFFER_ENABLED_KEY, true);
+  }
+
+  @Test
+  public void testSlowLogFilters() throws Exception {
```

Comment:
I think we need some coverage for multi threaded ringbuffer access (which is very common)..